src/main/java/org/sagebionetworks/bridge/fitbit/bridge/FitBitUserIterator.java

```diff
@@ -0,0 +1,102 @@
+package org.sagebionetworks.bridge.fitbit.bridge;
+
+import java.io.IOException;
+import java.util.Iterator;
+
+import org.sagebionetworks.bridge.fitbit.worker.Constants;
+import org.sagebionetworks.bridge.rest.ClientManager;
+import org.sagebionetworks.bridge.rest.api.ForWorkersApi;
+import org.sagebionetworks.bridge.rest.model.ForwardCursorStringList;
+import org.sagebionetworks.bridge.rest.model.OAuthAccessToken;
+
+/** Helper class to abstract away Bridge's paginated API for OAuth tokens. */
+public class FitBitUserIterator implements Iterator<FitBitUser> {
+    // Package-scoped for unit tests
+    static final int PAGESIZE = 10;
+
+    // Instance invariants
+    private final ClientManager bridgeClientManager;
+    private final String studyId;
+
+    // Instance state tracking
+    private ForwardCursorStringList healthCodeList;
+    private int nextIndex;
+
+    /**
+     * Constructs a FitBitUserIterator for the given Bridge client and study. This kicks off requests to load the first
+     * page.
+     */
+    public FitBitUserIterator(ClientManager bridgeClientManager, String studyId) {
+        this.bridgeClientManager = bridgeClientManager;
+        this.studyId = studyId;
+
+        // Load first page. Pass in null offsetKey to get the first page.
+        loadNextPage(null);
+    }
+
+    // Helper method to load the next page of users, using the offsetKey to request the next page. Pass in null to get
+    // the first page.
+    private void loadNextPage(String offsetKey) {
+        // Call server for the next page.
+        try {
+            healthCodeList = bridgeClientManager.getClient(ForWorkersApi.class).getHealthCodesGrantingOAuthAccess(
+                    studyId, Constants.FITBIT_VENDOR_ID, PAGESIZE, offsetKey).execute().body();
+        } catch (IOException ex) {
+            // Iterator can't throw exceptions. Wrap in a RuntimeException.
+            throw new RuntimeException(ex);
+        }
+
+        // Reset nextIndex.
+        nextIndex = 0;
+    }
```

Comment:
Just an idea for down the road: it might be nice to build a generic Iterator class in the rest client library for all forward only cursor lists.